docs/rbe-platforms.md

```diff
@@ -262,14 +262,32 @@ timeouts. Timeouts are specified as durations like `1s`, `15m`, or `1h`.
 
 ### Remote persistent worker properties
 
-Similar to the local execution environment, the remote execution environment may also retain a long-running process acting as a [persistent worker](https://bazel.build/remote/persistent) to help reduce the total cost of cold-starts for build actions with high startup cost.
-
-To use remote persistent workers, the action must have `"recycle-runner": "true"`
-in `exec_properties`.
-
-The Bazel's flag ["--experimental_remote_mark_tool_inputs"](https://bazel.build/reference/command-line-reference#flag--experimental_remote_mark_tool_inputs) should help set these automatically so you don't have to set them manually. However, we provide these `exec_properties` for rules authors to experiment with.
-
-- `persistentWorkerKey`: unique key for the persistent worker. This should be automatically set by Bazel.
+BuildBuddy's remote execution environment supports a feature called
+**remote persistent workers,** which are the remote equivalent of Bazel's
+local [persistent workers](https://bazel.build/remote/persistent).
+
+Persistent workers help reduce the total cost of cold starts for build
+actions. They are often used to improve the performance of build actions
+running on the JVM (such as Java or Kotlin compile actions), which are
+slow to run initially, but then get faster as Just-In-Time (JIT)
+compilation kicks in and allows the compiler to run using more optimized
+machine code.
+
+To use remote persistent workers, two steps are required:
+
+1. Set `"recycle-runner": "true"` in `exec_properties` to enable
+   [runner recycling](#action-isolation-and-hermeticity-properties).
+   We recommend setting this property using a Bazel macro, so that
```

Comment:
This seems pretty annoying. Have you tried `--modify_execution_info`? I'll take a look tomorrow, there has to be a better way than macro (if there isn't, we need to add one).